WeScan/Scan/RectangleFeaturesFunnel.swift

```diff
@@ -86,10 +97,16 @@ final class RectangleFeaturesFunnel {
             return
         }
         
-        if let previousRectangle = currentRectangle,
-            bestRectangle.rectangleFeature.isWithin(matchingThreshold, ofRectangleFeature: previousRectangle) {
+        if let previousRectangle = currentRectangle, bestRectangle.matchingScore >= minNumberOfMatches && bestRectangle.rectangleFeature.isWithin(10.0, ofRectangleFeature: previousRectangle) {
+            
+            thresholdPassed += 1
+            if thresholdPassed > autoScanThreshold {
+                thresholdPassed = 0
```

Comment:
When is the `thresholdPassed` getting reset? 
If it's never reset, we could have an instance where it gets really close to pass, then the user moves the camera, and on the next match it will pass (autoscan) which is not good.